### ISSUE TITLE
[X-VLA] Poor spatial precision and background sensitivity after fine-tuning with 150 episodes

### ISSUE THREAD

**Issue by triet4p** [huggingface/lerobot#3535]
### Ticket Type

❓ Technical Question

### Environment & System Info

```Shell
- LeRobot version: 0.5.1
- Platform: Windows-10-10.0.19045-SP0
- Python version: 3.12.13
- Huggingface Hub version: 1.10.1
- Transformers version: N/A
- Datasets version: N/A
- Numpy version: 2.2.6
- FFmpeg version: N/A
- PyTorch version: 2.10.0+cpu
- Is PyTorch built with CUDA support?: False
- Cuda version: N/A
- GPU model: N/A
- Using GPU in script?: <fill in>
- lerobot scripts: ['lerobot-calibrate', 'lerobot-dataset-viz', 'lerobot-edit-dataset', 'lerobot-eval', 'lerobot-find-cameras', 'lerobot-find-joint-limits', 'lerobot-find-port', 'lerobot-imgtransform-viz', 'lerobot-info', 'lerobot-record', 'lerobot-replay', 'lerobot-setup-can', 'lerobot-setup-motors', 'lerobot-teleoperate', 'lerobot-train', 'lerobot-train-tokenizer']
```

### Description

I'm fine-tuning X-VLA on a coffee-pouring task ("pick up cup X → pour into cup Y") and hitting two distinct failure modes. Looking for advice on data scale and collection strategy.

---

**Setup**
- Robot: SO-ARM-101
- Camera rig: top-view (main) + wrist cam (auxiliary)
- Training data: 150 episodes × 30–45 sec each
- Fine-tuning approach: Freeze Vision Encoder and Language Encoder, finetune only Soft-prompt and Transformers. *Use specific domain id (15)*

**Failure modes**

*1. Imprecise pouring — misses cup Y*
The robot picks up cup X and pours, but the trajectory doesn't align accurately with cup Y. This suggests the model lacks sufficient spatial precision for the final alignment phase, possibly due to limited variation in cup Y positions across training episodes.

*2. Background sensitivity — fails to pick up cup X on new tables*
Performance degrades significantly when the background changes. This is a classic generalization failure, likely caused by the model overfitting to table texture/color features rather than learning object-centric representations.

---

**My questions**
1. What episode count is typically needed to cover a task like this reliably with X-VLA?
2. Beyond raw count, what diversity axes matter most — backgrounds, lighting, cup positions, or cup appearances?
3. Any specific augmentation or data collection strategies that have worked for you on similar contact-rich manipulation tasks?
4. Can I finetune using existed domain id of XVLA instead of new domain id?

I suspect the background sensitivity issue in particular requires either more diverse backgrounds in training data, or domain randomization during collection. But I'm unsure whether to prioritize quantity or diversity given limited recording time.

Thanks in advance!

### Relevant logs or stack trace

```Shell

```

### Checklist

- [x] I have searched existing tickets to ensure this isn't a duplicate.
- [ ] I am using the latest version of the `main` branch.
- [x] I have verified this is not an environment-specific problem.

**Comment by jashshah999** [comment]
Poor spatial precision after fine-tuning can be a data quality issue. [lerobot-doctor](https://github.com/jashshah999/lerobot-doctor) checks training readiness including:

- Action clipping (values hitting limits = lost precision)
- Zero-variance dimensions (useless features confusing the policy)
- Normalization readiness (zero-std dims that cause NaN)
- Stuck actuators (bad demonstrations contaminating training)

```bash
lerobot-doctor /path/to/your/150_episodes --checks training,actions,anomalies
```

https://github.com/jashshah999/lerobot-doctor

**Comment by jashshah999** [comment]
Update: [lerobot-doctor](https://github.com/jashshah999/lerobot-doctor) can now score each episode and identify the bad ones:

```bash
lerobot-doctor score /path/to/your/150_episodes --json
# Outputs: per-episode quality score + recommended drops
```

Also `lerobot-doctor gate --policy smolvla` validates training compatibility before you start.

https://github.com/jashshah999/lerobot-doctor
